drivers/gles2/rasterizer_canvas_gles2.cpp

```diff
@@ -543,13 +543,104 @@ static const GLenum gl_primitive[] = {
 
 void RasterizerCanvasGLES2::_canvas_item_render_commands(Item *p_item, Item *current_clip, bool &reclip, RasterizerStorageGLES2::Material *p_material) {
 
+	// We will keep track of what state we have set OpenGL to throughout rendering commands,
+	// in order to minimize OpenGL state changes. This struct keeps a record of state and performs
+	// housekeeping.
+	struct StateCanvasItem {
```

Comment:
Move this struct definition to header file

Open to discussion, but I think the name "CommandState" would be clearer

Reply:
Yep that should be better. :+1: